index.js

```diff
@@ -1053,9 +1066,10 @@
                       constraints,
                       expArgTypes.concat([expRetType]),
                       values[idx],
-                      index,
+                      isNaN(index) ? expArgTypes.length : index,
```

Comment:
Actually, passing around `expTypes` rather than `expArgTypes` and `expRetType` turns out to be simpler. `NaN` is no longer necessary. :)